tokio/src/macros/cfg.rs

```diff
@@ -481,6 +481,36 @@ macro_rules! cfg_not_has_atomic_u64 {
     }
 }
 
+macro_rules! cfg_has_const_mutex_new {
+    ($($item:item)*) => {
+        $(
+            #[cfg(all(
+                not(all(loom, test)),
+                any(
+                    feature = "parking_lot",
+                    not(tokio_no_const_mutex_new)
+                )
+            ))]
+            $item
+        )*
+    }
+}
+
+macro_rules! cfg_not_has_const_mutex_new {
+    ($($item:item)*) => {
+        $(
+            #[cfg(not(all(
+                not(all(loom, test)),
+                any(
+                    feature = "parking_lot",
+                    not(tokio_no_const_mutex_new)
+                )
+            )))]
```

Comment:
nit, take it or leave it: would it make sense to simplify this by just having the build script not set the `tokio_no_const_mutex_new` cfg when the `loom` and `test` cfgs are set? or is this simpler?

Reply:
I believe this is the simplest option.